primitives/utility/src/lib.rs

```diff
@@ -51,3 +61,191 @@ impl<T: UpwardMessageSender, W: WrapVersion> SendXcm for ParentAsUmp<T, W> {
 		}
 	}
 }
+
+/// Contains information to handle refund/payment for xcm-execution
+#[derive(Clone, Eq, PartialEq, Debug)]
+struct AssetTraderRefunder<Balance, AssetId> {
+	// The amount of weight bought minus the weigh already refunded
+	weight: Weight,
+	// The balance paid for the weight bought minus the balance already refunded
+	balance: Balance,
+	// The multilocation representing the fungible asset in which fees were paid
+	location: MultiLocation,
```

Comment:
Is this representing the `MultiLocation` of the asset, or the location of the purchaser? If it's the former, that is already covered by the `AssetId` down below. If it's the latter, then this field should be renamed as `purchaser`.

Reply:
Alright what I did instead is store everything as a MultiAsset, now the struct contains only two fields: weight and outstanding_concrete_asset.